Backend2.2

areshkk opened the pull request:
## Для новых функций:
✨ Добавлена система аутентификации пользователей
🎭 Реализовано отслеживание настроения с 8 эмоциями
📊 Добавлена панель управления с историей настроений

## Для улучшений безопасности:
🔒 Улучшена валидация паролей и хеширование
🛡️ Добавлена защита от SQL-инъекций
✅ Реализована проверка сложности паролей

## Для исправлений ошибок:
🐛 Исправлена ошибка валидации email
🔧 Исправлена работа базы данных
🚑 Критическое исправление безопасности

## Для обновлений документации:
📝 Обновленный README утвержден всеми участниками
🔍 Добавлены комментарии в код
📋 Обновлена документация API
Frontend выполнен в ветке frontend_sprint2

millemir committed:
Доп проверка

Добавлены новые файлы и изменены старые

areshkk committed:
Создание и "оживление интерфейса"

Функциональный сайт, исправлены ошибки

millemir reviewed (approved):
Какая красота!

annawwee reviewed (approved):
Все молодцы

Laggon commented on Техническое_задание.md:
Изменения в техническое задание стоит оформлять отдельным PR и также запрашивать у всех ревью

Laggon reviewed (changes requested):
Начиная со второго спринта у всех должны быть индивидуальные задачи и кто-то один ответственный (одна задача и PR были выполнены в этом спринте, какая третья задача планируется? добавьте ее в issues)

При отправки формы регистрации выдает ошибку

```
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: user
[SQL: SELECT user.id AS user_id, user.username AS user_username, user.email AS user_email, user.password_hash AS user_password_hash, user.created_at AS user_created_at 
FROM user 
WHERE user.username = ?
 LIMIT ? OFFSET ?]
[parameters: ('admin', 1, 0)]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
```

Если нужно создать базу данных вручную, то стоит указать как это сделать в README.md

Laggon commented:
Как вариант можно разбить эту задачу на две, если это соответствует техническому заданию

millemir committed:
Настроены формы регистрации и входа

- Полная система аутентификации пользователей
- Отслеживание настроения с 8 эмоциями
- Панель управления с историей записей
- Система безопасности с валидацией паролей
- Современный UI с Bootstrap 5
- Документация и инструкции по запуску

annawwee reviewed (approved):
молодцы мы

Laggon reviewed (changes requested):
При отправки формы регистрации все также выдает ошибку

Вижу дополнили инструкцию по установке, но не подтянули в эту ветку, для этого стоит выполнить [слияние изменений из основной ветки](https://github.com/Laggon/docs/blob/main/merge/index.md)

И дополнить инструкцию по созданию базы данных в отдельном PR с ревью, а потом также подтянуть в эту ветку

Название PR должно отражать суть изменений (обычно совпадает с названием задачи)

Laggon commented:
Ответственным должен быть кто-то один
